cmd/integrationArtifactUpload.go

```diff
@@ -0,0 +1,205 @@
+		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error: %v", httpMethod, uploadIflowStatusURL, responseBody)
+	}
+	return errors.Errorf("Failed to create Integration Flow artefact, Response Status code: %v", uploadIflowStatusResp.StatusCode)
+}
+
+//UpdateIntegrationArtifact - Update existing integration artifact
+func UpdateIntegrationArtifact(config *integrationArtifactUploadOptions, httpClient piperhttp.Sender, fileUtils piperutils.FileUtils) error {
+	httpMethod := "POST"
+	header := make(http.Header)
+	header.Add("content-type", "application/json")
+	updateIflowStatusURL := fmt.Sprintf("%s/api/v1/IntegrationDesigntimeArtifactSaveAsVersion?Id='%s'&SaveAsVersion='%s'", config.Host, config.IntegrationFlowID, config.IntegrationFlowVersion)
+	payload, jsonError := GetJSONPayloadAsByteArray(config, "update", fileUtils)
+	if jsonError != nil {
+		return errors.Wrapf(jsonError, "Failed to get json payload for file %v, failed with error", config.FilePath)
+	}
+	updateIflowStatusResp, httpErr := httpClient.SendRequest(httpMethod, updateIflowStatusURL, payload, header, nil)
+
+	if updateIflowStatusResp != nil && updateIflowStatusResp.Body != nil {
+		defer updateIflowStatusResp.Body.Close()
+	}
+
+	if updateIflowStatusResp == nil {
+		return errors.Errorf("did not retrieve a HTTP response: %v", httpErr)
+	}
+
+	if updateIflowStatusResp.StatusCode == http.StatusOK {
+		log.Entry().
+			WithField("IntegrationFlowID", config.IntegrationFlowID).
+			Info("Successfully updated integration flow artefact in CPI designtime")
+		return nil
+	}
+	if httpErr != nil {
+		responseBody, readErr := ioutil.ReadAll(updateIflowStatusResp.Body)
+		if readErr != nil {
+			return errors.Wrapf(readErr, "HTTP response body could not be read, Response status code: %v", updateIflowStatusResp.StatusCode)
+		}
+		log.Entry().Errorf("a HTTP error occurred! Response body: %v, Response status code: %v", responseBody, updateIflowStatusResp.StatusCode)
+		return errors.Wrapf(httpErr, "HTTP %v request to %v failed with error: %v", httpMethod, updateIflowStatusURL, responseBody)
+	}
+	return errors.Errorf("Failed to update Integration Flow artefact, Response Status code: %v", updateIflowStatusResp.StatusCode)
+}
+
+//GetJSONPayloadAsByteArray -return http payload as byte array
+func GetJSONPayloadAsByteArray(config *integrationArtifactUploadOptions, mode string, fileUtils piperutils.FileUtils) (*bytes.Buffer, error) {
+	fileContent, readError := fileUtils.FileRead(config.FilePath)
+	if readError != nil {
+		return nil, errors.Wrapf(readError, "Error reading file")
+	}
+	jsonObj := gabs.New()
+	if mode == "create" {
+		jsonObj.Set(config.IntegrationFlowID, "Name")
+		jsonObj.Set(config.IntegrationFlowID, "Id")
+		jsonObj.Set(config.PackageID, "PackageId")
+		jsonObj.Set(b64.StdEncoding.EncodeToString(fileContent), "ArtifactContent")
```

Comment:
[minor] maybe it would be better to have the line for the `ArtifactContent` outside the `if`-`else` since that line is the same on both branches.